src/control/cmd/daos_agent/fabric.go

```diff
@@ -39,10 +40,20 @@ func (f *FabricInterface) String() string {
 	return fmt.Sprintf("%s%s (%s)", f.Name, dom, netdetect.DevClassName(f.NetDevClass))
 }
 
-// DefaultFabricInterface is the one used if no devices are found on the system.
-var DefaultFabricInterface = &FabricInterface{
-	Name:   "lo",
-	Domain: "lo",
+// AddProvider adds a provider to the FabricInterface.
+func (f *FabricInterface) AddProvider(provider string) {
+	if f == nil || provider == "" {
+		return
+	}
+
+	for _, p := range f.Providers {
+		// Avoid adding duplicates
+		if p == provider {
+			return
+		}
+	}
```

Comment:
Not sure how this is used; I don't see any other reference to it. You could use a map for uniqueness instead though, and build a slice of keys on demand for anyone who wants the list of providers for an interface.

Reply:
Just laying the groundwork for future improvements. Figured if we're condensing duplicate interfaces, we may as well save what providers they support. This code will need to be more provider-aware as time goes forward anyway.